### PR TITLE
Updating polkadotApi to work with asset V2

### PR DESCRIPTION
**Pull request by sumitsnk**
This PR resolves the ticket SNO-197.

### CODE REVIEW

File: src/net/polkadot.ts [hunk 1]
```diff
@@ -85,26 +85,24 @@ export default class Polkadot extends Api {
...
         }
-
-        let ethAssetID = polkadotApi.createType('AssetId', 'ETH');
-        // create asset ID for tokens
+        account = await polkadotApi.query.assets.account(PARACHAIN_ETHER_ASSET_ID, polkadotAddress);
         if (isErc20(asset)) {
-          ethAssetID = polkadotApi.createType('AssetId', { Token: asset.address });
+          let assetId: any = await polkadotApi.query.erc20App.assetId(asset.address);
+          if (assetId.isNone) {
+            throw new Error('Asset Not found');
```
[the remark this answers is not in the document]
Comment (sumitsnk) [reply src/net/polkadot.ts:98]: Do we need to return balance as zero if the assetId is missing? Also should we go ahead and use tuple now ?
As it needs changes to other files on a relative basis.

[the remark this answers is not in the document]
Comment (sumitsnk) [reply src/net/polkadot.ts]: ok, I have performed changes as per suggestion.

File: src/net/polkadot.ts [hunk 2]
```diff
@@ -85,26 +85,24 @@ export default class Polkadot extends Api {
   ): Promise<string> {
     try {
       if (polkadotApi) {
```
[the remark this answers is not in the document]
Comment (vgeddes) [reply src/net/polkadot.ts]: Awesome! Thanks.

**Review by vgeddes — APPROVED** [review]
Looks good now. Approved.

File: src/net/polkadot.ts [hunk 3]
```diff
@@ -82,32 +84,24 @@ export default class Polkadot extends Api {
...
-          );
-
-          return balance.toString();
+      let account: any;
+      // check if the asset is the native DOT asset and return DOT balance
+      if (isDot(asset)) {
+        const balance = await Polkadot.getGasCurrencyBalance(polkadotApi, polkadotAddress);
+        return { isAssetFound:false, amount:balance };
+      }
+      account = await polkadotApi.query.assets.account(PARACHAIN_ETHER_ASSET_ID, polkadotAddress);
```
Comment (vgeddes) [inline src/net/polkadot.ts:95]: This code flow is still a bit confusing though, and we end up making two calls to `query.assets.account` if the asset is erc20, which is unnecessary.

So can we separate the body into into clear distinct cases using `if` statements:
- if DOT, fetch dot balances
- if Ether, fetch ether balances
- If Erc20, erc20 balances

**Review by vgeddes — CHANGES_REQUESTED** [review]

File: src/net/polkadot.ts (hunk 3 above)
Comment (sumitsnk) [reply src/net/polkadot.ts]: As suggested, I've modified the code.

**Review by alistair-singh — APPROVED** [review]

**Review by vgeddes — APPROVED** [review]
Looks good now. thanks.

**Pull request merged**